### PR TITLE
Fixed Access Group Problems. SPC-42 #resolve

### PR DESCRIPTION
**Pull request by cfscosta**

### CODE REVIEW

File: src/main/java/org/fenixedu/spaces/domain/accessControl/SpecificSpaceManagersGroup.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright © 2002 Instituto Superior Técnico
+ *
+ * This file is part of FenixEdu Academic.
```
Comment (pedrosan7os) [inline src/main/java/org/fenixedu/spaces/domain/accessControl/SpecificSpaceManagersGroup.java]: wrong

File: src/main/java/org/fenixedu/spaces/domain/accessControl/SpecificSpaceManagersGroup.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+import org.fenixedu.bennu.core.annotation.GroupOperator;
+import org.fenixedu.bennu.core.domain.User;
+import org.fenixedu.bennu.core.groups.DynamicGroup;
+import org.fenixedu.bennu.core.groups.GroupStrategy;
+import org.fenixedu.bennu.core.i18n.BundleUtil;
+import org.fenixedu.spaces.domain.Space;
+import org.joda.time.DateTime;
+
+@GroupOperator("specificSpaceManagersGroup")
```
Comment (pedrosan7os) [inline src/main/java/org/fenixedu/spaces/domain/accessControl/SpecificSpaceManagersGroup.java]: This is 'specific'. should I expect a 'generic' space managers group in the near future? *sarcasm face*

Also, the group suffix is a little redundant. just call it spaceManager

File: src/main/java/org/fenixedu/spaces/domain/accessControl/SpecificSpaceManagersGroup.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    @Override
+    public Set<User> getMembers() {
+        return Space.getSpaces().filter(Space::isActive).map(s -> s.getManagementGroup()).filter(g -> g != null)
+                .flatMap(g -> g.getMembers().stream()).collect(Collectors.toSet());
+    }
+
+    @Override
+    public boolean isMember(User user) {
+        if (DynamicGroup.get("spaceManager").isMember(user)) {
```
Comment (pedrosan7os) [inline src/main/java/org/fenixedu/spaces/domain/accessControl/SpecificSpaceManagersGroup.java]: This is a terrible mixup. This group should be limited to the semantic of being a managers of some space. It is also inconsistent with getMembers()

**Comment by pedrosan7os** [comment]
This issue is a bug, however the solutions adds another group to the API. Also, 'Fixed Access Group Problems' is a little... please

File: src/main/java/org/fenixedu/spaces/domain/accessControl/SpecificSpaceManagersGroup.java (hunk 2 above)
Comment (cfscosta) [inline src/main/java/org/fenixedu/spaces/domain/accessControl/SpecificSpaceManagersGroup.java]: it is called 'specific' to follow the naming convention of existing groups that mimic this behavior and usage

File: src/main/java/org/fenixedu/spaces/domain/accessControl/SpaceManagersGroup.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    @Override
+    public boolean isMember(User user, DateTime when) {
+        return isMember(user);
+    }
+
+    @Override
+    public String getPresentationName() {
+        final String name =
+                BundleUtil.getString(getPresentationNameBundle(), getPresentationNameKey(), getPresentationNameKeyArgs());
```
Comment (pedrosan7os) [inline src/main/java/org/fenixedu/spaces/domain/accessControl/SpaceManagersGroup.java]: Just implement the presentation name without auxiliary methods. The place from where you copied had this like that so they can be extended as needed, keeping a uniform convention. In this case, you are outside of academic, and this class is not intended for extension.

**Pull request merged**